src/test/scala/RegexTokenizerTest.scala

```diff
@@ -0,0 +1,182 @@
+    sparkSession.sparkContext.setLogLevel("ERROR")
+    val encoder = RowEncoder.apply(testSchema)
+    val rowMemoryStream = new MemoryStream[Row](1,sqlContext)(encoder)
+
+    var rowDataset = rowMemoryStream.toDF
+    val regex: String = "\\d+"
+    // create Scala udf for tokenizer
+    val regexUDF = functions.udf(new RegexTokenizerUDF, DataTypes.createArrayType(DataTypes.BinaryType, false))
+    // register tokenizer udf
+    sparkSession.udf.register("regex_udf", regexUDF)
+    // create pattern col for regex
+    rowDataset = rowDataset.withColumn("regex", functions.lit(regex))
+    // apply tokenizer udf to column
+    rowDataset = rowDataset.withColumn("tokens", regexUDF.apply(functions.col("_raw"), functions.col("regex")))
+
+    // create Scala udf for ByteArrayListasStringList
+    val byteArrayListAsStringListUDF = functions.udf(new ByteArrayListAsStringListUDF, DataTypes.createArrayType(StringType))
+    sparkSession.udf.register("bytes_to_string_udf", byteArrayListAsStringListUDF)
+    rowDataset = rowDataset.withColumn("tokensAsStrings", byteArrayListAsStringListUDF.apply(functions.col("tokens")))
+
+    val streamingQuery = startStream(rowDataset)
+    var run: Long = 0
+
+    while (streamingQuery.isActive) {
+      val time = Timestamp.valueOf(LocalDateTime.ofInstant(Instant.now, ZoneOffset.UTC))
+      rowMemoryStream.addData(
+        makeRows(time, String.valueOf(run)))
+
+      run += 1
+
+      if (run == 10) {
+        streamingQuery.processAllAvailable
+        streamingQuery.stop
+        streamingQuery.awaitTermination()
+      }
+    }
+
+    val resultCollected = sqlContext.sql("SELECT tokensAsStrings FROM TokenAggregatorQuery").collect()
+
+    for (row <- resultCollected) {
+      val tokens = row.getAs[mutable.WrappedArray[String]]("tokensAsStrings")
+
+      assert(tokens.contains("00257"))
+      assert(tokens.contains("2022"))
+      assert(tokens.contains("52362"))
+      assert(tokens.contains("320000"))
+      assert(!tokens.contains("127.127"))
+      assert(!tokens.contains("=0"))
+      assert(!tokens.contains(".127 "))
+      assert(!tokens.contains("10:13:40"))
+      assert(!tokens.contains("service=tcp/port:8151"))
+      assert(!tokens.contains("duration="))
+      assert(!tokens.contains("fox"))
+
+    }
+  }
+
+  private def makeRows(time: Timestamp, partition: String): Seq[Row] = {
+
+    val rowList: ArrayBuffer[Row] = new ArrayBuffer[Row]
+
+    for (i <- 0 until amount.toInt) {
+      rowList += Row(
+        time,
+        exampleString,
+        "topic",
+        "stream",
+        "host",
+        "input",
+        partition,
+        0L,
+        exampleString.length.toLong
+      )
+    }
+    rowList
+  }
+
+  private def generateRawData(): Array[String] = {
+    val testDataList = new Array[String](amount.toInt)
+
+    for (i <- testDataList.indices) {
+      val randomVal = Math.floor(Math.random() * 999)
+      val text = "ip=172.17.255."+randomVal+",port=8080,session_id=46889"
```

Comment:
IP range won't be valid most of the time as the range is 0-255, but replace the header to `127.0.0.` anyways

Reply:
this helper method was actually not used so I removed it